modules/xmpp/SdpTransformUtil.js

```diff
@@ -0,0 +1,439 @@
+import * as transform from 'sdp-transform';
+
+/**
+ * Parses the primary SSRC of given SSRC group.
+ * @param {object} group the SSRC group object as defined by the 'sdp-transform'
+ * @return {Number} the primary SSRC number
+ */
+export function parsePrimarySSRC(group) {
+    return parseInt(group.ssrcs.split(" ")[0]);
+}
+
+/**
+ * Parses the secondary SSRC of given SSRC group.
+ * @param {object} group the SSRC group object as defined by the 'sdp-transform'
+ * @return {Number} the secondary SSRC number
+ */
+export function parseSecondarySSRC(group) {
+    return parseInt(group.ssrcs.split(" ")[1]);
+}
+
+function findGroup(mLine, groupSemantics, ssrcs) {
+    return mLine.ssrcGroups && mLine.ssrcGroups.find(
+            (group) => group.semantics === groupSemantics
+                            && !ssrcs || ssrcs === group.ssrcs);
+}
+
+function findGroupByPrimarySSRC(mLine, groupSemantics, primarySsrc) {
+    return mLine.ssrcGroups && mLine.ssrcGroups.find(
+            (group) => group.semantics === groupSemantics
+                            && parsePrimarySSRC(group) === primarySsrc);
+}
+
+function getSSRCCount(mLine) {
+    if (!mLine.ssrcs) {
+        return 0;
+    } else {
+        return mLine.ssrcs
+            .map(ssrcInfo => ssrcInfo.id)
+            .filter((ssrc, index, array) => array.indexOf(ssrc) === index)
+            .length;
+    }
+}
+
+/**
+ * Utility class for SDP manipulation using the 'sdp-transform' library.
+ *
+ * Typical use usage scenario:
+ *
+ * const transformer = new SdpTransformWrap(rawSdp);
+ * if (transformer.selectMedia('video)) {
+ *     transformer.addSSRCAttribute({
+ *         id: 2342343,
+ *         attribute: "cname",
+ *         value: "someCname"
+ *     });
+ *     rawSdp = transformer.toRawSdp();
+ * }
+ */
+export class SdpTransformWrap {
```

Comment:
i'm not a fan of this as an instance class that takes in the raw sdp, i think it would be better to have this be a set of static utility functions that take in an sdp and perform an operation (more of a functional-type approach).  i think it also becomes clearer in the methods that need to do something for them to take the sdp directly and call, for example, `SdpUtil.addRtxSsrc(mLine, rtx ssrc info...)` then `sdpTransformWrap.addRtxSsrc(rtx ssrc info)` because it's a bit confusing which sdp is being modified (since it's hidden).  i think mainly i find the obfuscation of the underlying sdp by the wrapper a bit weird, but maybe i'm making a big deal out of nothing, so open to discussing it.